npc_simulator/CMakeLists.txt

```diff
@@ -36,24 +45,62 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   )
 ament_export_dependencies(rosidl_default_runtime)
 
-ament_auto_add_library(${PROJECT_NAME}_node STATIC src/node.cpp)
+###
+# create library
+###
+add_library(${PROJECT_NAME}_node STATIC src/node.cpp)
 target_link_libraries(${PROJECT_NAME}_node "${Boost_LIBRARY_DIR}")
-ament_target_dependencies(${PROJECT_NAME}_node Boost)
+target_include_directories(${PROJECT_NAME}_node PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BIN_DIR}/rosidl_generator_cpp>
+  $<INSTALL_INTERFACE:include>
+  )
 
-ament_auto_add_executable(${PROJECT_NAME}_main
-  src/main.cpp
+# capitalize the boost dependency
+set(node_deps ${${PROJECT_NAME}_BUILD_DEPENDS})
+list(REMOVE_ITEM node_deps boost)
+list(APPEND node_deps Boost)
+
+ament_target_dependencies(${PROJECT_NAME}_node
+  ${node_deps}
 )
+
 # to allow using interfaces created in this package
 rosidl_target_interfaces(${PROJECT_NAME}_node
   ${PROJECT_NAME} "rosidl_typesupport_cpp"
 )
 
+# export targets because rosidl automatically exports targets so relying on old-style variables fails in client
+# packages; e.g. the include directories for dependencies of this packages are not taken over. So we cannot use
+# `ament_auto_package()`
+ament_export_targets(export_${PROJECT_NAME}_node HAS_LIBRARY_TARGET)
+
+###
+# create executable
+###
+ament_auto_add_executable(${PROJECT_NAME}_main
+  src/main.cpp
+)
+target_link_libraries(${PROJECT_NAME}_main ${PROJECT_NAME}_node)
+
+
 if(BUILD_TESTING)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
 endif()
 
-# export information to downstream packages
-ament_auto_package(
-  INSTALL_TO_SHARE launch
+# install(
+#   DIRECTORY include/
```

Comment:
Why is this commented out?

Reply:
Ah, okay, the question should probably be why the `install(TARGETS` below _isn't_ commented out since `ament_auto_package` does that usually – can you add a comment for that if it's not by accident?

Reply:
@nnmm thanks for spotting the commented code, I removed it. I checked the source of `ament_auto_package()`, it doesn't install the exported target